meson.build

```diff
@@ -259,6 +252,7 @@ conf.set_quoted('DATADIR', datadir)
 conf.set_quoted('LOCALSTATEDIR', localstatedir)
 conf.set_quoted('SYSCONFDIR', sysconfdir)
 conf.set_quoted('PLUGINDIR', plugin_dir)
+conf.set_quoted('FWUP_EFI_DIR_NAME', 'fedora') # FIXME
```

Comment:
I think this wouldn't actually be configured at compile time.  If we're bringing fwup in tree, I would rather see it be an entirely runtime decision based on `/etc/lsb-release`.  It would make snap and flatpak more flexible too then (assuming fwupd managed the asset and put it in place)

Reply:
Assuming we ship fwupd-efi in-tree, I agree. Otherwise we'd obviously need to set FWUP_EFI_DIR_NAME to wherever fwup.efi has been installed...

Reply:
Well I think even if it's not in tree, I would rather see it stored in somewhere predictable like:
/usr/lib/<triplet>/<subdirectory>/fwup<arch>.efi

Make distros no longer responsible to install in packaging, but instead have fwupd put it in place where it should go based on whatever lsb-release is showing and whatever ESP mount location is configured.

I'm sick of all the failures from not installing the asset to the right place, and that would put an end to it entirely.

Reply:
100% agree. I'll admit I looked a bit at how fwup-efi is built, and I ran away.

Reply:
I think it looks scarier than it is.  At the end of the day it's building a shared library that isn't installed and then using `objcopy` to copy out the relevant bits to make an EFI application and `readelf` to strip out the build ID.

The part that makes it more precarious is the zillions of flags passed in to build the shared library and which ones are actually important.

Reply:
If you have a spare few cycles I'd really appreciate the help converting it to use meson. I'm happy to carry on with fixing up the dell plugin to create the fake esrt entry, and I'm also slowly adding unit tests (finding bugs everywhere!) to the code we've imported.

Reply:
Unit tests are awesome.  I'll be glad we won't randomly have breakage in UEFI plugin in the future now because of them 👍 

I think on Monday I should be able to put aside some cycles to sort the EFI application.  Fortunately it looks like systemd guys have a nice meson setup for building an EFI application that can be leveraged and tweaked.

Reply:
Well I managed to get it building., just pushed.
No idea if it works, that will be for `$FUTURE` checking.